### PR TITLE
add parsing custom data

### LINKED ISSUE #9: Format URL to show clickable 'website' button instead of whole url
This could be handled it three ways:
1. Frontend should scan for 'http' in data and then replace content with 'website' word. Drawback - if some data would have two websites, every would be treated same way
2. Frontend could get some specific formatting, e.g. besides data, backend could also serve format, or data type. It could be sth like that `{data:{name: point1, website: {type: hyperlink, value: www...}}}` or sth like makrdown `{data:{name: point1, website: (website)[www...]}`
3. Frontend can serve only one webpage and it has reserved keyword for value 'www' or 'website'.

Choose one or suggest another approach.

### PR DESCRIPTION
**Pull request by kacper-cyra**
So I went with little modified second approach. Backend can store and send user data as a dictionary. The dictionary would consist of three fields, required: `type` (ex. "hyperlink"), `value` (ex. "https://www.google.com"); and optional: `displayValue` (a string that if exists will be displayed instead of value).
Custom values are recognizable and parsed by `type` filed. If `type` is not recognizable `displayValue` or `value` will be displayed as string.

Waiting for feedback 😄 

Backend changes: https://github.com/Problematy/goodmap/pull/21

Closes #9

### CODE REVIEW

**Comment by raven-wing** [comment]
I didn't tell this, but initially I had in mind custom types which casts to specific way of displaying, so (if I understand this code correctly) is great.
As for javascript code I'd like someone else to review this changes.

From my point of view I have one little thing which is not yet common for this repository, but would be nice. It is unit test. :)

File: tests/popup_data.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import "@testing-library/jest-dom";
+
+// import * as React from "react";
```
Comment (raven-wing) [inline tests/popup_data.test.js:3]:
```suggestion
```

**Review by raven-wing — APPROVED** [review]

File: src/popup_data.js [hunk 2]
```diff
@@ -1,29 +1,62 @@
...
+        </p>
+      );
+    }
+
+    return (
+      <p key={dataKey} className="m-0">
+        <b>{dataKey}</b>
+        {": " + getContentAsString(value)}
+      </p>
+    );
```
Comment (KrystianKjjk) [inline src/popup_data.js:48]: In this case - u can create variable which stores CustomDataComponent or getContentAsString(value) depends on condition and then pass this variable to the jsx, so you won't duplicate it

Comment (raven-wing) [reply src/popup_data.js:48]:
```suggestion
    const DataComponent = isCustomValue(value) ? mapCustomTypeToReactComponent(value) : getContentAsString(value)
  
    return (
      <p key={dataKey} className="m-0">
        <b>{dataKey}</b>
        {": "}
        {DataComponent}
      </p>
    );
  };

```

Comment (raven-wing) [reply src/popup_data.js:48]: @KrystianKjjk sth like that, right?

Comment (kacper-cyra) [reply src/popup_data.js:48]: I have already refactored it to separate files in https://github.com/Problematy/goodmap-frontend/issues/18 as it's branch is based on this branch.

Comment (kacper-cyra) [reply src/popup_data.js:48]: Improved in [PR](https://github.com/Problematy/goodmap-frontend/pull/25/files#diff-ae78d83a350f971de8f0339822120adae40a4a033e98e400d7b1c62d0d80038e)

**Pull request merged**